scripts/end-to-end/verify-testcases.py

```diff
@@ -0,0 +1,174 @@
+
+        print(self.file + ":", len(self.tests), "test-cases.")
+
+        self.ready = True
+
+    @staticmethod
+    def parse_parameters(line, trace):
+        input = re.search(r'\s*in:\s*([a-fA-F0-9]*)', line, re.M | re.I)
+        if input:
+            trace.input = input.group(1)
+        output = re.search(r'\s*out:\s*([a-fA-F0-9]*)', line, re.M | re.I)
+        if output:
+            trace.output = output.group(1)
+        result = re.search(r'\s*result:\s*([a-fA-F0-9]*)', line, re.M | re.I)
+        if result:
+            trace.result = result.group(1)
+        gas_used = re.search(r'\s*gas\sused:\s*([a-fA-F0-9]*)', line, re.M | re.I)
+        if gas_used:
+            trace.gas = gas_used.group(1)
+        value = re.search(r'\s*value:\s*([a-fA-F0-9]*)', line, re.M | re.I)
+        if value:
+            trace.value = value.group(1)
+
+    def diff(self, analyser):
+        if not self.ready:
+            self.analyse()
+        if not analyser.ready:
+            analyser.analyse()
+
+        intersection = set(self.tests.keys()) & set(analyser.tests.keys())
+        mismatches = set()
+
+        for test_name in intersection:
+            left = self.tests[test_name]
+            right = analyser.tests[test_name]
+            if len(left.traces) != len(right.traces):
+                mismatches.add((test_name, "trace count are different: " + str(len(left.traces)) +
+                                " != " + str(len(right.traces))))
+            else:
+                self.check_traces(test_name, left, right, mismatches)
+
+        for mismatch in mismatches:
+            print(mismatch[0])
+            print(mismatch[1])
+
+        print(len(intersection), "test-cases - ", len(mismatches), " mismatche(s)")
+
+    def check_traces(self, test_name, left, right, mismatches):
+        for trace_id in range(0, len(left.traces)):
+            left_trace = left.traces[trace_id]
+            right_trace = right.traces[trace_id]
+            assert (left_trace.kind == right_trace.kind)
+            if str(left_trace) != str(right_trace):
+                mismatch_info = "    " + str(left_trace) + "\n"
+                mismatch_info += "    " + str(right_trace) + "\n"
+                mismatch_info += "    "
+                for ch in range(0, len(str(left_trace))):
+                    if ch < len(str(left_trace)) and ch < len(str(right_trace)):
+                        if str(left_trace)[ch] != str(right_trace)[ch]:
+                            mismatch_info += "|"
+                        else:
+                            mismatch_info += " "
+                    else:
+                        mismatch_info += "|"
+                mismatch_info += "\n"
+                mismatches.add((test_name, mismatch_info))
+
+
+def main():
+    base_path = os.path.dirname(__file__)
+    semantic_trace = TraceAnalyser(base_path + "/semanticTests.trace")
+    end_to_end_trace = TraceAnalyser(base_path + "/SolidityEndToEndTest.trace")
```

Comment:
Maybe it's nicer to have the trace files as arguments instead of hard-coded paths relative to the script directory.